core/store/src/trie/mod.rs

```diff
@@ -1252,11 +1253,12 @@ impl Trie {
     /// `charge_gas_for_trie_node_access` is used to control whether Trie node
     /// accesses incur any gas. Note that access to values is never charged here;
     /// it is only charged when the returned ref is dereferenced.
```

Comment:
nit: Could we update the description of this function to specify what `map_result` is and how is it used different to retrieve the value vs check existence?

Reply:
Sure, can do. Although it feels somewhat weird to document internal helper functions to that level of detail, given that the code provides the same answer in a more concise, precise and formal way.

Reply:
Upto you, no worries!

Reply:
I think I managed to strike a good balance in the most recent commit.